plugin/storage/kafka/options.go

```diff
@@ -69,12 +75,30 @@ func (opt *Options) AddFlags(flagSet *flag.FlagSet) {
 		defaultEncoding,
 		fmt.Sprintf(`(experimental) Encoding of spans ("%s" or "%s") sent to kafka.`, EncodingJSON, EncodingProto),
 	)
+	flagSet.Int(
+		configPrefix+suffixRequiredAcks,
+		defaultRequiredAcks,
+		"(experimental) Required kafka broker acknowledgement. default = 1, no response = 0, wait for local = 1, wait for all = -1",
+	)
+	flagSet.Int(
+		configPrefix+suffixCompression,
+		defaultComression,
+		"(experimental) Type of compression to use on messages. default = 0, none = 0, gzip = 1, snappy = 2, lz4 = 3, zstd = 4",
+	)
+	flagSet.Int(
+		configPrefix+suffixCompressionLevel,
+		defaultCompressionLevel,
```

Comment:
As mentioned [here](https://github.com/jaegertracing/jaeger/pull/1524#discussion_r285469314), you need to specify the default for the flag based on the `compression` value that has been defined by the user. Otherwise, regardless of which `compression` option has been chosen, the level will always default to `-1000`.

This also means you should have a specific test for each of the defaults.

Reply:
I don't think the flags package provides a way to define the default as a function so probably best to keep it returning the `-1000` default, and use this as an indication that a compression specific default needs to be calculated. So in `kafka/producer/config.go` have something like:
```
saramaConfig.Producer.CompressionLevel = getCompressionLevel(c.CompressionLevel, saramaConfig.Producer.Compression)
```